>From edc0e9499f80e5d0d91c6cfa77c63f91928f7a48 Mon Sep 17 00:00:00 2001
From: cixi <gengcx@spreadst.com>
Date: Wed, 27 Feb 2019 13:07:04 +0800
Subject: [PATCH] Fix open function: mode argument must be supplied when
 O_CREAT is specified in flags

Signed-off-by: cixi <gengcx@spreadst.com>
---
 testcases/ddt/alsa_test_suite/src/testcases/st_alsa_capture_file_test.c | 2 +-
 .../ddt/filesystem_test_suite/src/testcases/st_filesystem_copy_file.c   | 2 +-
 .../filesystem_test_suite/src/testcases/st_filesystem_write_to_file.c   | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/testcases/ddt/alsa_test_suite/src/testcases/st_alsa_capture_file_test.c b/testcases/ddt/alsa_test_suite/src/testcases/st_alsa_capture_file_test.c
index 258d238..809edb1 100644
--- a/testcases/ddt/alsa_test_suite/src/testcases/st_alsa_capture_file_test.c
+++ b/testcases/ddt/alsa_test_suite/src/testcases/st_alsa_capture_file_test.c
@@ -83,7 +83,7 @@ int st_audio_file_capture(tc_dev_params * pcm_param)
 	loops = total_size / size;
 
 	fd_audio =
-	    open((const char *)(pcm_param->file_name), O_WRONLY | O_CREAT);
+	    open((const char *)(pcm_param->file_name), O_WRONLY | O_CREAT,0666);
 	if (-1 == fd_audio) {
 		TEST_PRINT_ERR("file open failed ");
 		rc = FAILURE;
diff --git a/testcases/ddt/filesystem_test_suite/src/testcases/st_filesystem_copy_file.c b/testcases/ddt/filesystem_test_suite/src/testcases/st_filesystem_copy_file.c
index 21c5d2b..137b19e 100644
--- a/testcases/ddt/filesystem_test_suite/src/testcases/st_filesystem_copy_file.c
+++ b/testcases/ddt/filesystem_test_suite/src/testcases/st_filesystem_copy_file.c
@@ -89,7 +89,7 @@ int st_filesystem_performance_copy_test(struct st_filesystem_testparams *info,
 			srcfileopenflag = 1;
 			goto free_mem;
 		}
-		dstfdes = open((const char *)dstfile_ptr, (O_WRONLY | O_CREAT));
+		dstfdes = open((const char *)dstfile_ptr,O_WRONLY | O_CREAT,0666);
 		if (-1 == dstfdes) {
 			perror("\n open");
 			TEST_PRINT_ERR("dst file open failed ");
diff --git a/testcases/ddt/filesystem_test_suite/src/testcases/st_filesystem_write_to_file.c b/testcases/ddt/filesystem_test_suite/src/testcases/st_filesystem_write_to_file.c
index 80e8e24..e194dc2 100644
--- a/testcases/ddt/filesystem_test_suite/src/testcases/st_filesystem_write_to_file.c
+++ b/testcases/ddt/filesystem_test_suite/src/testcases/st_filesystem_write_to_file.c
@@ -89,7 +89,7 @@ int st_filesystem_performance_write_test(struct st_filesystem_testparams *info,
 	}
 	TEST_PRINT_TRC("mmap srcfile ");
 
-	fdes = open((const char *)file_ptr, O_WRONLY | O_CREAT);
+	fdes = open((const char *)file_ptr, O_WRONLY | O_CREAT,0666);
 	if (-1 == fdes) {
 		perror("\n open");
 		TEST_PRINT_ERR("file open failed ");
-- 
1.9.1

